cpp/CMakeLists.txt

```diff
@@ -138,46 +94,20 @@ function(cpp_cc_enable_clang_formatting)
   mark_as_advanced(
     ${CODING_CONV_PREFIX}_ClangFormat_OPTIONS ${CODING_CONV_PREFIX}_ClangFormat_FILES_RE
     ${CODING_CONV_PREFIX}_ClangFormat_EXCLUDES_RE)
-  execute_process(
-    COMMAND git ls-files --error-unmatch .clang-format
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    RESULT_VARIABLE is_clang_format_config_tracked
-    ERROR_QUIET)
-  if(NOT is_clang_format_config_tracked EQUAL 0)
-    if(EXISTS ${PROJECT_SOURCE_DIR}/.clang-format.changes)
-      cpp_cc_find_python_module(yaml REQUIRED)
-      if(YAML_FOUND)
-        execute_process(
-          COMMAND
-            ${PYTHON_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/cpplib.py" merge-yaml
-            "${${CODING_CONV_PREFIX}_ClangFormat_CONFIG_FILE}"
-            "${PROJECT_SOURCE_DIR}/.clang-format.changes" "${PROJECT_SOURCE_DIR}/.clang-format"
-          RESULT_VARIABLE merge_yaml_status)
-        if(NOT ${merge_yaml_status} EQUAL 0)
-          message(SEND_ERROR "Error occured while merging Clang-Format YAML configurations")
-        endif()
-      else()
-        message(FATAL_ERROR "Cannot find Python yaml module, which is needed to merge YAML files.")
-      endif()
-    else()
-      configure_file(${${CODING_CONV_PREFIX}_ClangFormat_CONFIG_FILE}
-                     ${PROJECT_SOURCE_DIR}/.clang-format COPYONLY)
-    endif()
-  endif()
-  unset(is_clang_format_config_tracked)
 
+  cpp_cc_setup_tool_config(clang-format ${ClangFormat_EXECUTABLE})
   set(clang_format_command_prefix
       ${PYTHON_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/bbp-clang-format.py"
       -S "${PROJECT_SOURCE_DIR}" "--executable=${ClangFormat_EXECUTABLE}"
       --files-re ${${CODING_CONV_PREFIX}_ClangFormat_FILES_RE}
       --excludes-re ${${CODING_CONV_PREFIX}_ClangFormat_EXCLUDES_RE})
-  add_custom_target(clang-format ${clang_format_command_prefix} --action format --
+  add_custom_target(clang-format ${clang_format_command_prefix} format --
                                  "${${CODING_CONV_PREFIX}_ClangFormat_OPTIONS}")
-  add_custom_target(check-clang-format ${clang_format_command_prefix} --action check --
+  add_custom_target(check-clang-format ${clang_format_command_prefix} check --
                                        "${${CODING_CONV_PREFIX}_ClangFormat_OPTIONS}")
   if(${CODING_CONV_PREFIX}_TEST_FORMATTING)
     add_test(NAME ClangFormat
-             COMMAND ${clang_format_command_prefix} --action check --make-unescape-re --
+             COMMAND ${clang_format_command_prefix} --make-unescape-re check --
```

Comment:
I prefer to distinguish the sub-command options from the tool global options, for instance:

```bash
git --git-dir=/path/to/repo checkout -b branch_name
```

That being said, having CLI options accepting an infinite number of arguments confuses argparse in the following case:
```bash
$ ./bbp-clang-format.py --files-re foo bar check
[...]
bbp-clang-format.py: error: the following arguments are required: action, options
```
because `check` is sucked into `files-re`.

`--` is required to instruct argparse the option argument is finished
```bash
bbp-clang-format --files-re foo bar -- check
```

I will make the required change in the CMakeLists.txt, and properly test it!
Thanks

Reply:
tackled in [beb3d84](https://github.com/BlueBrain/hpc-coding-conventions/pull/123/commits/beb3d847145045352fc2269c22b7e8d2d0dcae53)